Admin/Remove-DuplicateEntriesFromIanaMappings.ps1

```diff
@@ -0,0 +1,205 @@
+
+            Write-Host "[+] Duplicate entries detected!" -ForegroundColor Yellow
+
+            try {
+                $writeProgressParams.Activity = $activityBase + " Creating backup $mappingBackupFilePath"
+                Write-Progress @writeProgressParams
+
+                # If duplicate entries were detected, create a backup of the file before modifying it
+                Copy-Item -Path $mappingFilePath -Destination $mappingBackupFilePath
+                Write-Host "[+] Backup created: $mappingBackupFilePath"
+            } catch {
+                Write-Host "[+] Failed to create backup file. Inner Exception: $_" -ForegroundColor Red
+                Write-Progress @writeProgressParams -Completed
+
+                return
+            }
+
+            $dupeIndex = 0
+
+            # Iterate through all of the duplicate entries and remove them one by one
+            foreach ($dupe in $duplicateEntries) {
+                Write-Host "[+] Processing duplicate entry: $dupe"
+
+                $writeProgressParams.Activity = $activityBase + " Processing duplicate entry: $dupe"
+                Write-Progress @writeProgressParams
+
+                try {
+                    # Select the duplicate node and remove it - we use SelectSingleNode here as it could be that there are multiple duplicates
+                    $singleNode = $ianaMappingXml.SelectSingleNode("//Map[@IANA='$($dupe.IANA)' and @Win='$($dupe.Win)']")
+                    $singleNode.ParentNode.RemoveChild($singleNode) | Out-Null
+
+                    $dupeIndex++
+                } catch {
+                    Write-Host "[+] Failed to fix duplicate entry $dupe. Inner Exception: $_" -ForegroundColor Red
+                    Write-Progress @writeProgressParams -Completed
+
+                    return
+                }
+            }
+
+            # Validate that all duplicate entries were removed before saving the mapping file
+            if ($duplicateEntries.Count -eq $dupeIndex) {
+                Write-Host "[+] All duplicate entries were removed" -ForegroundColor Green
+
+                $writeProgressParams.Activity = $activityBase + " Saving changes to file IanaTimeZoneMappings.xml"
+                Write-Progress @writeProgressParams
+
+                try {
+                    # Save the modified IanaTimeZoneMappings.xml file and override the existing one
+                    $ianaMappingXml.Save($mappingFilePath)
+
+                    # Restart services if the -RestartServices $true was used when running the script - otherwise do nothing
+                    if ($PerformServiceRestart) {
+                        Write-Host "[+] Restart services: W3SVC, WAS & MSExchangeTransport"
+
+                        $writeProgressParams.Activity = $activityBase + " Restarting services W3SVC, WAS & MSExchangeTransport"
+                        Write-Progress @writeProgressParams
+
+                        Restart-Service -Name W3SVC, WAS, MSExchangeTransport -Force
```

Comment:
Can this throw an exception?

Reply:
There is a chance. Will place this into its own try/catch block just to make sure. Resolved with next commit.